docs/changelog/96928.yaml

```diff
@@ -0,0 +1,5 @@
+pr: 96928
+summary: Remove invalid asserts in `MergingDigest`
+area: Aggregations
+type: bug
+issues: []
```

Comment:
I don't think this change needs a changelog entry. 
It is not a bug in a released version of Elasticsearch.